File: baekjoon/2839.cpp
/*
    2839 : 설탕 배달
    URL : https://www.acmicpc.net/problem/2839
    Input :
        18
    Output :
        4
*/
#include <iostream>

using namespace std;

int main()
{
    int N;
    cin >> N;

    int five = (N / 5);
    int three = (N % 5) / 3;

    do
    {
        if ((5 * five) + (3 * three) == N)
        {
            cout << (five + three);
            return 0;
        }
        else
        {
            five = five - 1;
            three = (N - (5 * five)) / 3;
        }
    } while (five >= 0);
    cout << -1;

    return 0;
}

File: baekjoon/2839.py
"""
    2839 : 설탕 배달
    URL : https://www.acmicpc.net/problem/2839
    Input :
        18
    Output :
        4
"""
N = int(input())

five = int(N // 5)
three = int((N % 5) // 3)

solve = False
while True:
    if (5 * five) + (3 * three) == N:
        print(five + three)
        solve = True
        break
    else:
        five = five - 1
        three = (N - (5 * five)) // 3

    if five < 0:
        break;

if not solve:
    print(-1)